### ISSUE TITLE
Question/Feature Request: Explicit Config, DefaultConfig.Instance & COMPLUS_TieredCompilation

### ISSUE THREAD

**Issue by yahorsi** [dotnet/BenchmarkDotNet#1297]
Hi Guys,

I'm trying to run the benchmark on the 3.0 and tiered jit matters, so, I'd like to make sure all benchmarks run in tier 1 and so I have to set COMPLUS_TieredCompilation. In order to do so I use following config:
```CSharp
    class Program
    {
        private const string JitTieredCompilation = "COMPLUS_TieredCompilation";

        static void Main(string[] args)
        {
            var config = DefaultConfig.Instance
                .With(Job.Default.With(CoreRuntime.Core30).WithEnvironmentVariable(JitTieredCompilation, "1"));

            BenchmarkSwitcher
                .FromAssembly(typeof(Program).Assembly)
                .Run(args, config);
        }
    }
```

The problem is that using such config I actually have 2 different jobs, 1 that is "default" without COMPLUS_TieredCompilation set and another one with COMPLUS_TieredCompilation  set. I didn't find a simple way to:
1. Get a config with just 1 job with COMPLUS_TieredCompilation set
2. Get ManualConfig instance configured right

So, once 3.0 has tiered jit it would be cool to:
1. Have some simple shortcut to run all benchmarks in tier N
2. Have some default config without any jobs set

**Comment by yahorsi** [comment]
Update, the problem was caused by how I was running benchmarks.
I had config made in code and in the same time 2 was running benchmarks using
```
dotnet run -c Release -f netcoreapp3.0 --runtimes netcoreapp3.0 --filter *
```
that, as I understand, adds "default" job

It is still very confusing how to run benchmarks on the multiple frameworks, what command-line switch --runtimes means and what it affects and how it related to the config mane in code. And even more, we have a dedicated tool to run benchmarks.

**Comment by adamsitnik** [comment]
Hi @yahorsi 

Please excuse me for a very big delay.

When BDN is merging the config provided programmatically and via console line args there are some situations where it makes imperfect decisions.

Since you are not the first person that run into similar issue when benchmarking multiple runtimes, I've written a doc that explains how to do it properly.

https://benchmarkdotnet.org/articles/configs/toolchains.html#multiple-frameworks-support

TL;DR: Your config should avoid setting the runtime in explicit way:

```diff
- Job.Default.With(CoreRuntime.Core30).WithEnvironmentVariable(JitTieredCompilation, "1")
+ Job.Default.WithEnvironmentVariable(JitTieredCompilation, "1")
```

And you should set it only via the console line argument (if the target runtime must be different than the runtime of the host process).

Moreover, to make sure that BDN is going to extend given `Job` by the settings from console line args, you need to use the `job.AsDefault()` extension method. This is recognized by BDN as "base settings" that just needs to be extended by console line settings. By default, it just adds new jobs.

```cs
var config = DefaultConfig.Instance
    .With(Job.Default.WithEnvironmentVariable(JitTieredCompilation, "1").AsDefault());
```

Some time ago we have added the possibility to specify env vars via console line args, so in this case you can use console args only:

```cs
class Program
{
    static void Main(string[] args) => BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args);
}
```

To run against the $TFM runtime:

```cmd
dotnet run -c Release -f $TFM --envVars COMPLUS_TieredCompilation:1
```

To run against multiple runtimes:

```cmd
dotnet run -c Release -f $TFM --envVars COMPLUS_TieredCompilation:1 --runtimes $TFM1 $TFM2
```

Thanks,
Adam
